package.json

```diff
@@ -19,7 +19,7 @@
   "main": "./lib",
   "dependencies": {
     "buffer-writer": "1.0.1",
-    "generic-pool": "2.1.1",
+    "generic-pool": "git://github.com/Shyp/node-pool.git#shyp-master",
```

Comment:
Hopefully I'll get it merged this weekend! I haven't really had a chance to read it all through yet. Just put out a bunch of minor bug fixes and stuff in v2.4.1 so these fixes should be in v2.5.0 and the only thing I have lined up for it